src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -182,10 +184,31 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
 
             searchOptions.UnsupportedSearchParams = unsupportedSearchParameters;
 
-            // Sort is currently not implimented. The sort parameters are added to the search options to allow for future development.
-            if (searchParams.Sort != null && searchParams.Sort.Count > 0)
+            if (searchParams.Sort?.Count > 0)
             {
-                searchOptions.Sort = searchParams.Sort.Select(sorting => Tuple.Create(sorting.Item1, (SortOrder)sorting.Item2));
+                var sortings = new List<(SearchParameterInfo, SortOrder)>();
+                List<(string parameterName, string reason)> unsupportedSortings = null;
```

Comment:
Why initialize the list to null and then do null checks on it on line 201 and 206 instead of initializing it to an empty list?

Reply:
Because we can save an allocation when all parameters are properly specified.